Cargo.toml

```
@@ -62,10 +62,12 @@ asm = ["sha2/asm"]
 batch = ["alloc", "merlin", "rand_core"]
 fast = ["curve25519-dalek/precomputed-tables"]
 digest = ["signature/digest"]
-# This features turns off stricter checking for scalar malleability in signatures
+# Turns off stricter checking for scalar malleability in signatures
 legacy_compatibility = []
 pkcs8 = ["ed25519/pkcs8"]
 pem = ["alloc", "ed25519/pem", "pkcs8"]
 rand_core = ["dep:rand_core"]
+# Exposes the unsafe raw_sign() and raw_sign_prehashed() functions
+raw_sign = []
```

The `cfg!()` calls use `raw-sign`, and I think the dash is a bit nicer than the underscore for feature flags.

```suggestion
raw-sign = []
```

---

I'd suggest calling this feature `hazmat` to match a `hazmat` module.

Also in addition to having these kind of low-level signing functions, it'd be nice to have low-level verifying functions as well, which also permit customizing the digest for people who want to do that.

---

What extra knobs would raw verification have? I looked at BIP32 and its verification algorithm is unchanged from the original spec. I believe FROST's is also equivalent to the original spec (after the correct pubkey has been computed).

---

Oh you mean make  the non-prehashed variant to generic over a 512-bit hash function? Is this something people asked for?

---

Yes, e.g. #164

---

Ah ok. Then maybe `raw_sign` should be generic too?

---

Looks like it's now generic around `CtxDigest` so we should be good